operators/src/processing/temporal_raster_aggregation/min_max_first_last_subquery.rs

```diff
@@ -0,0 +1,375 @@
+
+impl NoDataIgnoringAccFunction for MinIgnoreNoDataAccFunction {
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if value == no_data {
+                return acc;
+            } else if acc == no_data {
+                return value;
+            }
+        }
+
+        if acc < value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+pub struct MaxAccFunction {}
+
+impl AccFunction for MaxAccFunction {
+    fn acc<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if acc == no_data || value == no_data {
+                return no_data;
+            }
+        }
+
+        if acc > value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+pub struct MaxIgnoreNoDataAccFunction {}
+
+impl NoDataIgnoringAccFunction for MaxIgnoreNoDataAccFunction {
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if value == no_data {
+                return acc;
+            } else if acc == no_data {
+                return value;
+            }
+        }
+
+        if acc > value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+pub struct LastValidAccFunction {}
+
+impl NoDataIgnoringAccFunction for LastValidAccFunction {
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if value == no_data {
+                return acc;
+            }
+        }
+        value
+    }
+}
+pub struct FirstValidAccFunction {}
+
+impl NoDataIgnoringAccFunction for FirstValidAccFunction {
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if acc == no_data {
+                return value;
+            }
+        }
+        acc
+    }
+}
+
+pub fn fold_fn<T, C>(
+    acc: TemporalRasterAggregationTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> TemporalRasterAggregationTileAccu<T>
+where
+    T: Pixel,
+    C: AccFunction,
+{
+    let mut accu_tile = acc.accu_tile;
+
+    let grid = if acc.initial_state {
+        tile.grid_array
+    } else {
+        match (accu_tile.grid_array, tile.grid_array) {
+            (GridOrEmpty::Grid(mut a), GridOrEmpty::Grid(g)) => {
+                a.data = a
+                    .inner_ref()
+                    .iter()
+                    .zip(g.inner_ref())
+                    .map(|(x, y)| C::acc(a.no_data_value, *x, *y))
+                    .collect();
+                GridOrEmpty::Grid(a)
+            }
+            (GridOrEmpty::Empty(e), _) | (_, GridOrEmpty::Empty(e)) => GridOrEmpty::Empty(e),
+        }
+    };
+
+    accu_tile.grid_array = grid;
+    TemporalRasterAggregationTileAccu {
+        accu_tile,
+        initial_state: false,
```

Comment:
Can't you use an empty grid as initial state?

Reply:
nope then you cant know if you already had an empty tile